controllers/user.go

```diff
@@ -111,6 +112,68 @@ func (c *ApiController) DeleteUser() {
 	c.ServeJSON()
 }
 
+// @Title ForgetPassword
+// @Description verify email or phoneId and change password
+// @Param   username    formData   string  true        "The username of the user"
+// @Param   password    formData   string  true        "The password of the user"
+// @Param   organization    formData   string  true        "The organization of the user"
+// @Param   email    formData   string  true        "The email of the user"
+// @Param   emailCode    formData   string  true        "The emailCode of the user"
+// @Success 200 {object} controllers.Response The Response object
+// @router /forget-password [post]
+func (c *ApiController) ForgetPassword() {
+	var resp Response
+
+	var form RequestForm
+	err := json.Unmarshal(c.Ctx.Input.RequestBody, &form)
+	if err != nil {
+		panic(err)
+	}
+
+	// check result through Email or Phone
+	if strings.Contains(form.Email, "@") {
+		checkResult := object.CheckVerificationCode(form.Email, form.EmailCode)
+		object.DisableVerificationCode(form.Email)
+		if len(checkResult) != 0 {
+			responseText := fmt.Sprintf("Email%s", checkResult)
+			c.ResponseError(responseText)
+			return
+		}
+	} else {
+		checkPhone := fmt.Sprintf("+%s%s", form.PhonePrefix, form.Phone)
+		checkResult := object.CheckVerificationCode(checkPhone, form.PhoneCode)
+		object.DisableVerificationCode(form.Phone)
```

Comment:
We should not disable the verification code until the reset action is successfully completed.